applications/StructuralMechanicsApplication/python_scripts/master_stiffness_process.py

```diff
@@ -0,0 +1,203 @@
+import KratosMultiphysics
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_analysis import StructuralMechanicsAnalysis
+from math import cos, sin 
+
+def Factory(parameters, model):
+    if not isinstance(parameters, KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return MasterStiffnessMatrixProcess(model, parameters["Parameters"])
+
+## All the processes python should be derived from "Process"
+class MasterStiffnessMatrixProcess(KratosMultiphysics.Process):
+    """
+    This process allows the user to obtain the Stiffness Matrices of two Master nodes related to two specific Slave surfaces.\n
+    Example:\n
+    Master = MasterStiffnessMatrixProcess(model, parameters)\n
+    Master.Run()\n
+
+    Public member variables:
+    model -- the container of the different model parts.
+    parameters -- Kratos parameters containing solver parameters.
+    """
+
+    def __init__(self, model, parameters):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver parameters.
+        """
+        
+        KratosMultiphysics.Process.__init__(self)
+        
+        default_parameters = KratosMultiphysics.Parameters("""{
+                "model_part_list": ["Surface_name_1","Surface_name_2"],
```

Comment:
absolutely cosmetic comment, but i would do it as 

~~~json
[
    { 
      "model_part_name"   : "Surface_name_1",
      "master_coordinates" : [0.0,0.0,0.0]
    },
   { 
      "model_part_name"   : "Surface_name_2",
      "master_coordinates" : [1.0,0.0,0.0]
    }
]
~~~

simply because this way you can read at one the information block you need.

other than that, well done!!

Reply:
well, i think you should also have the "relative_perturbation_eps" to be user controllable

~~~json
[
    { 
      "model_part_name"   : "Surface_name_1",
      "master_coordinates" : [0.0,0.0,0.0],
      "eps_perturbation" : 1e-5
    },
   { 
      "model_part_name"   : "Surface_name_2",
      "master_coordinates" : [1.0,0.0,0.0],
        "eps_perturbation" : 1e-5
    }
]
~~~

Reply:
I would like to leave the parameters as a list due to the logic of the algorithm, but please let me know if it is necessary and I change it.